Fix SpringerNatureWebFetcherTest

Fix the fetcher - see https://devdocs.jabref.org/code-howtos/fetchers.html for help

```
SpringerNatureWebFetcherTest > searchByQueryFindsEntry() FAILED
    org.opentest4j.AssertionFailedError at SpringerNatureWebFetcherTest.java:140
```

I'm unsure whether this belongs to here, but instead of creating a new issue, I'm adding the Java exception from the console logs that I can reproduce when I ask JR to get the fulltext via the entryeditor:

>  [JabRef CachedThreadPool] org.jabref.logic.importer.fetcher.SpringerNatureFullTextFetcher.findFullText()
WARN: SpringerLink API request failed: kong.unirest.core.UnirestException: java.net.ConnectException
	at unirest.java.core@4.10.0/kong.unirest.core.DefaultInterceptor.onFail(DefaultInterceptor.java:43)
	at unirest.java.core@4.10.0/kong.unirest.core.CompoundInterceptor.lambda$onFail$0(CompoundInterceptor.java:54)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:214)
	at java.base/java.util.Collections$2.tryAdvance(Collections.java:5182)
	at java.base/java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:147)
	at java.base/java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:588)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:574)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:560)
	at java.base/java.util.stream.FindOps$FindOp.evaluateSequential(FindOps.java:150)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:265)
	at java.base/java.util.stream.ReferencePipeline.findFirst(ReferencePipeline.java:683)
	at unirest.java.core@4.10.0/kong.unirest.core.CompoundInterceptor.onFail(CompoundInterceptor.java:56)
	at unirest.java.core@4.10.0/kong.unirest.core.java.JavaClient.request(JavaClient.java:80)
	at unirest.java.core@4.10.0/kong.unirest.core.BaseRequest.request(BaseRequest.java:357)
	at unirest.java.core@4.10.0/kong.unirest.core.BaseRequest.asJson(BaseRequest.java:241)
	at org.jabref.jablib/org.jabref.logic.importer.fetcher.SpringerNatureFullTextFetcher.findFullText(SpringerNatureFullTextFetcher.java:57)
	at org.jabref.jablib/org.jabref.logic.importer.FulltextFetchers.lambda$getCallable$0(FulltextFetchers.java:108)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:328)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
	at java.base/java.lang.Thread.run(Thread.java:1474)
Caused by: java.net.ConnectException
	at unirest.java.core@4.10.0/kong.unirest.core.java.JavaClient.request(JavaClient.java:72)
	... 8 more
Caused by: java.net.ConnectException
	at java.base/java.util.concurrent.CompletableFuture.uniHandle(CompletableFuture.java:955)
	at java.base/java.util.concurrent.CompletableFuture.uniHandleStage(CompletableFuture.java:971)
	at java.base/java.util.concurrent.CompletableFuture.handle(CompletableFuture.java:2399)
	at java.base/java.util.concurrent.CompletableFuture$UniCompose.tryFire(CompletableFuture.java:1171)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:531)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1794)
	at java.base/java.util.concurrent.CompletableFuture.completeAsync(CompletableFuture.java:2761)
	... 9 more
Caused by: java.nio.channels.UnresolvedAddressException
	at java.base/sun.nio.ch.Net.checkAddress(Net.java:144)
	at java.base/sun.nio.ch.Net.checkAddress(Net.java:152)
	at java.base/sun.nio.ch.SocketChannelImpl.checkRemote(SocketChannelImpl.java:918)
	at java.base/sun.nio.ch.SocketChannelImpl.connect(SocketChannelImpl.java:936)
	... 16 more

--- a/jablib/src/main/java/org/jabref/logic/importer/fetcher/ArXivFetcher.java
+++ b/jablib/src/main/java/org/jabref/logic/importer/fetcher/ArXivFetcher.java
@@ -344,7 +344,7 @@ public class ArXivFetcher implements FulltextFetcher, PagedSearchBasedFetcher, I
                                                                                        }, executor))
                                                                                .toList();
 
-            Collection<BibEntry> modifiedSearchResult = futureSearchResult.stream()
+            List<BibEntry> modifiedSearchResult = futureSearchResult.stream()
                                                                           .map(CompletableFuture::join)
                                                                           .toList();
 

--- a/jablib/src/main/java/org/jabref/logic/importer/fetcher/ArXivFetcher.java
+++ b/jablib/src/main/java/org/jabref/logic/importer/fetcher/ArXivFetcher.java
@@ -345,8 +345,8 @@ public class ArXivFetcher implements FulltextFetcher, PagedSearchBasedFetcher, I
                                                                                .toList();
 
             List<BibEntry> modifiedSearchResult = futureSearchResult.stream()
-                                                                          .map(CompletableFuture::join)
-                                                                          .toList();
+                                                                    .map(CompletableFuture::join)
+                                                                    .toList();
 
             return new Page<>(result.getQuery(), result.getPageNumber(), modifiedSearchResult);
         }

--- a/jablib/src/main/java/org/jabref/logic/importer/fetcher/AstrophysicsDataSystem.java
+++ b/jablib/src/main/java/org/jabref/logic/importer/fetcher/AstrophysicsDataSystem.java
@@ -278,7 +278,7 @@ public class AstrophysicsDataSystem
         }
         // This is currently just interpreting the complex query as a default string query
         List<String> bibCodes = fetchBibcodes(urlForQuery);
-        Collection<BibEntry> results = performSearchByIds(bibCodes);
+        List<BibEntry> results = performSearchByIds(bibCodes);
         return new Page<>(queryList.toString(), pageNumber, results);
     }
 

--- a/jablib/src/main/java/org/jabref/model/paging/Page.java
+++ b/jablib/src/main/java/org/jabref/model/paging/Page.java
@@ -1,5 +1,6 @@
 package org.jabref.model.paging;
 
+import java.util.ArrayList;
 import java.util.Collection;
 import java.util.Collections;
 import java.util.List;
@@ -8,19 +9,19 @@ public class Page<T> {
 
     private int pageNumber;
     private String query;
-    private Collection<T> content;
+    private List<T> content;
 
-    public Page(String query, int pageNumber, Collection<T> content) {
+    public Page(String query, int pageNumber, List<T> content) {
         this.query = query;
         this.pageNumber = pageNumber;
-        this.content = Collections.unmodifiableCollection(content);
+        this.content = Collections.unmodifiableList(new ArrayList<>(content));
     }
 
     public Page(String query, int pageNumber) {
         this(query, pageNumber, List.of());
     }
 
-    public Collection<T> getContent() {
+    public List<T> getContent() {
         return content;
     }
 

--- a/jablib/src/main/java/org/jabref/model/paging/Page.java
+++ b/jablib/src/main/java/org/jabref/model/paging/Page.java
@@ -1,7 +1,6 @@
 package org.jabref.model.paging;
 
 import java.util.ArrayList;
-import java.util.Collection;
 import java.util.Collections;
 import java.util.List;
 

--- a/jablib/src/test/java/org/jabref/logic/importer/fetcher/SpringerNatureWebFetcherTest.java
+++ b/jablib/src/test/java/org/jabref/logic/importer/fetcher/SpringerNatureWebFetcherTest.java
@@ -44,6 +44,22 @@ class SpringerNatureWebFetcherTest implements SearchBasedFetcherCapabilityTest, 
     @Test
     void searchByQueryFindsEntry() throws FetcherException {
 
+        BibEntry articleGenerativeAISolutions = new BibEntry(StandardEntryType.Article)
+                .withField(StandardField.AUTHOR, "Gheventer, Andre and do Amaral Gurgel, Patricia and Henrique Brito, Carlos and Maiani de Mello, Rafael and Rocha, Sabrina and Feitosa, Rodrigo and Horta Travassos, Guilherme")
+                .withField(StandardField.DATE, "2026-05-13")
+                .withField(StandardField.DOI, "10.1007/s11219-026-09754-7")
+                .withField(StandardField.FILE, ":http\\://link.springer.com/openurl/pdf?id=doi\\:10.1007/s11219-026-09754-7:PDF")
+                .withField(StandardField.ISSN, "0963-9314")
+                .withField(StandardField.JOURNAL, "Software Quality Journal")
+                .withField(StandardField.MONTH, "#may#")
+                .withField(StandardField.NUMBER, "2")
+                .withField(StandardField.PAGES, "1--37")
+                .withField(StandardField.PUBLISHER, "Springer")
+                .withField(StandardField.TITLE, "Generative AI solutions for software quality: Assessing industrial readiness")
+                .withField(StandardField.VOLUME, "34")
+                .withField(StandardField.YEAR, "2026")
+                .withField(StandardField.ABSTRACT, "Software quality is undergoing a profound transformation, driven by state-of-the-art research on the application of emerging technologies in software development processes. Specifically, the use of generative Artificial Intelligence (AI) may represent an opportunity to advance the state of practice in this domain. This study aims to assess the industrial readiness and availability of Generative AI-based solutions for software quality, classifying them according to ISO/IEC 25010 attributes and SDLC phases. An empirical assessment of the state of practice was conducted, employing a Rapid Multivocal Literature Review (RMLR) protocol as a data collection instrument to screen evidence from academic databases (Scopus) and grey literature (Google, GitHub, PapersWithCode). We identified 24 potentially usable solutions. However, the analysis reveals a low technological maturity, with most solutions being academic prototypes hampered by fundamental technical limitations and adoption challenges. These include the “last mile problem” in translating research prototypes into reliable, production-ready tools; the “strategic adoption dilemma” forcing practitioners to trade off between proprietary lock-in and high open-source infrastructure costs; and the “scarcity of realistic public data,” which drives a generalization gap due to reliance on synthetic or leaked benchmarks. Generative AI in software quality remains an emerging but immature field, hampered by a critical reliability gap between academic prototypes and industrial needs. Advancing this domain requires moving beyond a narrow code-centric focus to address the quality of the AI systems themselves, expanding research across all SDLC phases and ISO 25010 attributes. We conclude with a roadmap advocating for contamination-free benchmarks, explainable architectures, and robust guidelines for real-world integration.");
+
         BibEntry articleSupportingIdentification = new BibEntry(StandardEntryType.Article)
                 .withField(StandardField.AUTHOR, "Iftikhar, Umar and Börstler, Jürgen and Bin Ali, Nauman and Kopp, Oliver")
                 .withField(StandardField.DATE, "2025-04-23")
@@ -138,7 +154,7 @@ class SpringerNatureWebFetcherTest implements SearchBasedFetcherCapabilityTest, 
                 .withField(StandardField.ABSTRACT, "Several Open-Source Software (OSS) projects depend on the continuity of their development communities to remain sustainable. Understanding how developers become inactive or why they take breaks can help communities prevent abandonment and incentivize developers to come back. In this paper, we propose a novel method to identify developers’ inactive periods by analyzing the individual rhythm of contributions to the projects. Using this method, we quantitatively analyze the inactivity of core developers in 18 OSS organizations hosted on GitHub. We also survey core developers to receive their feedback about the identified breaks and transitions. Our results show that our method was effective for identifying developers’ breaks. About 94% of the surveyed core developers agreed with our state model of inactivity; 71% and 79% of them acknowledged their breaks and state transition, respectively. We also show that all core developers take breaks (at least once) and about a half of them (~45%) have completely disengaged from a project for at least one year. We also analyzed the probability of transitions to/from inactivity and found that developers who pause their activity have a ~35 to ~55% chance to return to an active state; yet, if the break lasts for a year or longer, then the probability of resuming activities drops to ~21–26%, with a ~54% chance of complete disengagement. These results may support the creation of policies and mechanisms to make OSS community managers aware of breaks and potential project abandonment.");
 
         List<BibEntry> fetchedEntries = fetcher.performSearch("JabRef Social Barriers Steinmacher");
-        assertEquals(List.of(articleSupportingIdentification, articleTagThatIssue, fourthArticle, thirdArticle, firstArticle, secondArticle), fetchedEntries);
+        assertEquals(List.of(articleGenerativeAISolutions, articleSupportingIdentification, articleTagThatIssue, fourthArticle, thirdArticle, firstArticle, secondArticle), fetchedEntries);
     }
 
     @Test

--- a/jablib/src/test/java/org/jabref/model/paging/PageTest.java
+++ b/jablib/src/test/java/org/jabref/model/paging/PageTest.java
@@ -2,7 +2,6 @@ package org.jabref.model.paging;
 
 import java.util.ArrayList;
 import java.util.Arrays;
-import java.util.Collection;
 import java.util.Collections;
 import java.util.List;
 

--- a/jablib/src/test/java/org/jabref/model/paging/PageTest.java
+++ b/jablib/src/test/java/org/jabref/model/paging/PageTest.java
@@ -17,13 +17,13 @@ class PageTest {
     private Page<String> page2;
     private final int testPageNumber = 3;
     private final String testQuery = "anyQuery";
-    private Collection<String> testContent = new ArrayList<>();
+    private List<String> testContent = new ArrayList<>();
     private final String[] testStrings = {"str1", "str2", "str3"};
 
     @BeforeEach
     void setup() {
         testContent.addAll(Arrays.asList(testStrings));
-        testContent = Collections.unmodifiableCollection(testContent);
+        testContent = Collections.unmodifiableList(testContent);
         page1 = new Page<>(testQuery, testPageNumber, testContent);
         page2 = new Page<>(testQuery, testPageNumber);
     }

